src/Strategy/Python/Pip.hs

```diff
@@ -0,0 +1,126 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+
+module Strategy.Python.Pip (Package (..), PackageMetadata (..), getPackages, pipShowParser) where
+
+import Control.Effect.Diagnostics (Diagnostics, recover)
+import Data.Aeson (FromJSON, ToJSON)
+import Data.Functor (void)
+import Data.List (find)
+import Data.Maybe (mapMaybe)
+import Data.String.Conversion (toText)
+import Data.Text (Text)
+import Data.Text qualified as Text
+import Data.Void (Void)
+import Effect.Exec (
+  AllowErr (Never),
+  Command (..),
+  Exec,
+  Has,
+  execJson,
+  execParser,
+ )
+import GHC.Generics (Generic)
+import Path (Abs, Dir, Path)
+import Text.Megaparsec
+import Text.Megaparsec.Char (char)
+import Text.Megaparsec.Char.Lexer qualified as L
+
+type Parser = Parsec Void Text
+data InstalledPackage = InstalledPackage
+  { name :: Text
+  , version :: Text
+  }
+  deriving (Show, Eq, Ord, Generic, FromJSON)
+
+data PackageMetadata = PackageMetadata Text Text [Text]
+  deriving (Show, Eq, Ord)
+
+data Package = Package
```

Comment:
[Optional][Minor] use `PythonPackage` instead of `Package` since we are exporting this symbol.